packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -302,3 +368,53 @@ export async function visualizeSharedTreeNodeBySchema(
 		}
 	}
 }
+
+/**
+ * Creates a visual representation of a SharedTree based on its schema.
+ * @param tree - The {@link VerboseTree} to visualize
+ * @param treeDefinitions - The schema that defines the structure and types of the tree
+ * @param allowedTypes - String representing the allowed types for this node
+ * @param isRequired - Whether this field is required in its parent object schema.
```

Comment:
@alexvy86 @Josmithr 

Interested in your thoughts on this doc.

Reply:
I would probably note for `treeDefinitions` that it contains all of the definitions for the _entire_ tree, where `allowedTypes` describes what is allowed to go _here_ (if that makes sense).